### ISSUE TITLE
Fabio does not serve http2 with go >= 1.7

### ISSUE THREAD

**Issue by smancke** [eBay/fabio#215]
We found out, that fabio currently is not doing http2.
The reasons seems to be a change in go's behavior, when you supply a custom tls.Config (see https://github.com/golang/go/issues/15908).
Before 1.7 it was fine to have `NextProtos: nil`, now it has to be set explicitly. `NextProtos: []string{"h2"}`.
The same thing hit the caddy server (see https://github.com/mholt/caddy/pull/976).

When I add 
```go
	x := &tls.Config{
		NextProtos: []string{"h2"},
```
in cert/source.go it has http support. This seems to be consistent to the documentation of the method `shouldConfigureHTTP2ForServe()` in `net/http/server`.

**Comment by magiconair** [comment]
@smancke like this?

**Comment by smancke** [comment]
@magiconair yes

**Comment by magiconair** [comment]
I'm trying to see an HTTP/2 request in the `testSource()` function in `cert/source_test.go` but I don't see it. Trying to print the proto and somehow expecting `HTTP/2` here:

```
srv := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
	fmt.Println("Proto:", r.Proto)
	fmt.Fprint(w, "OK")
}))
```

Any idea what I'm missing?

**Comment by smancke** [comment]
the go http client does not do http2 by default, if you configure it with a custom tls config:

in net/http/transport.go
```go
func (t *Transport) onceSetNextProtoDefaults() {
...
	if t.TLSClientConfig != nil || t.Dial != nil || t.DialTLS != nil {
		// Be conservative and don't automatically enable
		// http2 if they've specified a custom TLS config or
		// custom dialers. Let them opt-in themselves via
		// http2.ConfigureTransport so we don't surprise them
		// by modifying their tls.Config. Issue 14275.
		return
	}
...
}
```

There must be a way to force the transport by hand, but I don't found how ..

**Comment by smancke** [comment]
@magiconair 
This works for configuring the client, using the golang/x http2 package:
```go
      import "golang.org/x/net/http2"

	clientTransport := &http.Transport{
		TLSClientConfig: &tls.Config{
			RootCAs: rootCAs,
		},
	}
	http2.ConfigureTransport(clientTransport)
	client := http.Client{
		Transport: clientTransport,
	}
```

Proto: HTTP/2.0

**Comment by magiconair** [comment]
@smancke Awesome. Thanks a lot. I've updated the change to include the test. Testing for HTTP/2 feels grafted on though. Maybe its better to move this to a separate test but for now at least there is a test.

**Comment by smancke** [comment]
Cool!!
Any idea, when this will be in a release?

**Comment by magiconair** [comment]
@smancke I can spin a 1.3.6 tomorrow. Also, time to wrap up the other lingering changes. Quick enough?

**Comment by smancke** [comment]
Yes, would be very great!
Thanks a lot!!

**Comment by magiconair** [comment]
Thanks for finding and helping! Appreciated.

**Comment by magiconair** [comment]
Merged to master and published in 1.3.6 release.
